packages/plugin-core/src/commands/GoToSiblingCommand.ts

```diff
@@ -33,65 +34,207 @@ export class GoToSiblingCommand extends BasicCommand<
+
+    // Check the date format for journal note. Only when date format of journal notes is default,
+    // navigate chronologically
+    const config = await engine.getConfig();
+    const dateFormat = config.data?.workspace.journal.dateFormat;
+    return dateFormat === "y.MM.dd";
+  }
+
+  private async getSiblingForJournalNote(
+    allNotes: NotePropsByIdDict,
+    currNote: NoteProps,
+    direction: Direction
+  ): Promise<{ sibling: NoteProps | null; msg: CommandOutput["msg"] | null }> {
+    const journalNotes = this.getSiblingsForJournalNote(allNotes, currNote);
+    // If the active note is the only journal note in the workspace, there is no sibling
+    if (journalNotes.length === 1) {
+      VSCodeUtils.showMessage(
+        MessageSeverity.WARN,
+        "There is no sibling journal note. Currently open note is the only journal note in the current workspace",
+        {}
+      );
+      return { sibling: null, msg: "no_siblings" };
+    }
+    // Sort all journal notes in the workspace
+    const sortedJournalNotes = _.sortBy(journalNotes, [
+      (note) => this.getDateFromJournalNote(note).valueOf(),
+    ]);
+    const currNoteIdx = _.findIndex(sortedJournalNotes, { id: currNote.id });
+    // Get the sibling based on the direction
+    let sibling: NoteProps;
+    if (direction === "next") {
+      sibling =
+        currNoteIdx !== sortedJournalNotes.length - 1
+          ? sortedJournalNotes[currNoteIdx + 1]
+          : sortedJournalNotes[0];
     } else {
-      const vault = PickerUtilsV2.getOrPromptVaultForOpenEditor();
-      const note = NoteUtils.getNoteByFnameFromEngine({
-        fname: value,
-        vault,
-        engine,
-      });
-      if (!_.isUndefined(note)) {
-        respNodes = engine.notes[note.parent as string].children
-          .map((id) => engine.notes[id])
-          .filter((ent) => _.isUndefined(ent.stub));
-      }
+      sibling =
+        currNoteIdx !== 0
+          ? sortedJournalNotes[currNoteIdx - 1]
+          : _.last(sortedJournalNotes)!;
+    }
+    return { sibling, msg: null };
+  }
+
+  private getSiblingsForJournalNote = (
+    notes: NotePropsByIdDict,
+    currNote: NoteProps
+  ): NoteProps[] => {
+    const month = notes[currNote.parent!];
+    const year = notes[month.parent!];
+    const parent = notes[year.parent!];
+    // Get all descendants of the parent. This includes stub notes
+    const allDescendants = this.getDescendantsRecursively(notes, parent);
+    // Filter out stub notes
+    return allDescendants.filter((note) => !note.stub);
+  };
+
+  private getDescendantsRecursively = (
+    notes: NotePropsByIdDict,
+    parent: NoteProps
+  ): NoteProps[] => {
+    // Get child notes
+    const children = parent.children.map((id) => notes[id]);
+    // Get all descendant notes of the given note, including stub notes
+    const descendants = [...children];
+    // Recursively get descendants
+    for (const child of children) {
```

Comment:
we don't want to descent indefinitely, just to cover year/month hierarchies. otherwise, notes like `daily.journal.2022.01.01.foo` would be caught in this filter as well